modules/test/playwright/tests/client-extension-web/editorConfigContributor.spec.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpers.fixture';
+import {clientExtensionsPageTest} from '../../fixtures/clientExtensionsPageTest';
+import {ClientExtensionsPage} from '../../pages/product-navigation-applications-menu/ClientExtensionsPage';
+
+export const test = mergeTests(apiHelpersTest, clientExtensionsPageTest);
+
+test('Editor Config Contributor CET is disabled if FF is set to false', async ({
+	_apiHelpers,
+	clientExtensionsPage,
+}: {
+	_apiHelpers: any;
+	clientExtensionsPage: ClientExtensionsPage;
+}) => {
+	await _apiHelpers.featureFlag.updateFeatureFlag('LPS-186870', false);
+
+	await clientExtensionsPage.page.reload();
+
+	await clientExtensionsPage.newClientExtensionButton.click();
+
+	const menuItemLocator = clientExtensionsPage.page.getByRole('menuitem', {
+		name: 'Add Editor Config Contributor',
+	});
+
+	await expect(menuItemLocator).not.toBeAttached();
+});
+
+test('Editor Config Contributor CET is enabled if FF is set to true', async ({
+	_apiHelpers,
+	clientExtensionsPage,
+}: {
+	_apiHelpers: any;
+	clientExtensionsPage: ClientExtensionsPage;
+}) => {
+	await _apiHelpers.featureFlag.updateFeatureFlag('LPS-186870', true);
+
+	await clientExtensionsPage.page.reload();
```

Comment:
I was considering adding this to a `beforeEach`. We might want to do that if there will be more tests in this file. This is something that we need to be careful about, since `updateFeatureFlag` changes state for all following tests.

We may also want to think of some global way to set FFs, as managing FFs with fetch requests will poorly affect performance, no matter how we do it.

Reply:
The FF problems would be avoided with the FF fixture.

However it's not clear how we could boost performance because we would need set the FF on and off per each test, unless the executor of the playwright tests (CI, user, etc.) guarantees that only tests using the same FFs combinations are accessing the test server. This could be done theoretically by inspecting the FF fixtures the test declare, but we are far from implementing it yet.

In any case, since it would be possible as a future optimization, I would not worry about performance now but prefer to do it "right" and evolvable.